### DISCUSSION TITLE
Pull Request | remote: Permission to arendst/Tasmota.git denied to ...

### DISCUSSION THREAD

**Discussion by mathias-buder** [arendst/Tasmota#17360] [Q&A] [answered]
Hi all,

I've developed a device driver that adds support for a narrow FoV ultrasonic sensor that I'll use in a water [level measurement system](https://github.com/mathias-buder/water-level-measure). However, when I try to create a pull-request against the developent branch (following the instruction [here](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md)) I receive message "remote: Permission to arendst/Tasmota.git denied to ..." as shown below.

Does anyone know what I did wrong?

Thanks!

![image](https://user-images.githubusercontent.com/34923690/207025559-8ff04151-e10e-4cdb-b9ec-d40616330c09.png)

**Comment by sfromis** [comment, marked as answer]
The message "Pushing to ..." looks like you are not merely trying to create a PR, but to actually push it to the repository, an action only a few people have permission to do. The normal flow would be for you instead to submit the PR for review and consideration.

**Reply by mathias-buder** [reply]
Thanks for the quick relay!

I've created a new branch (me007-ula-narrow-fov-ultrasonic-sensor) based off the development branch. I've tried to push my new branch and wanted to create a pull-request after that. Isn't that the correct process? This is the first time I do that ;-)

**Reply by sfromis** [reply]
I've not done a lot there, but what I do is to create my branch, and then create a pull requests for the project repository. When my change is reviewed and approved by one of the core developers, they are the ones to merge my PR.

Pushing is instead when you have the permission to update the repository at will, meaning that you need not go via the route of creating a PR for approval. For Tasmota that would only be the case for you if you are doing extensive work, and become one of the few core developers (which I'm not).

**Reply by joba-1** [reply]
No problem :)

Normal process for me is
* fork repo to my own github user
* clone the fork and checkout development branch
* modify, commit and push to the fork
* create a pull request from there (github will already offer that after a push)

**Reply by joba-1** [reply]
Since you have already done the work, there are a few options, one is

* fork tasmota to your github user
* clone the repo to a new directory and checkout development branch
* delete all files from that directory except .git/*
* copy all files except .git/ from the clone with your work to the new clones directory
* in the new directory do git status to see the changed files. If only your changes appear, add and commit them.
* push to your fork
* create pull request

hope it helps!

**Reply by mathias-buder** [reply]
Thanks a lot for all the replys.

I see now my mistake. I didn't fork the tasmota repository into my personal space. I actually never forked a repo before. However, once I did that everything worked just fine.
